aziotctl/src/config/apply.rs

```diff
@@ -37,10 +37,24 @@ pub(crate) fn run(options: Options) -> anyhow::Result<()> {
     // So when running as root, get the four users appropriately.
     // Otherwise, if this is a debug build, fall back to using the current user.
     // Otherwise, tell the user to re-run as root.
-    let aziotks_user = crate::internal::common::get_system_user("aziotks")?;
-    let aziotcs_user = crate::internal::common::get_system_user("aziotcs")?;
-    let aziotid_user = crate::internal::common::get_system_user("aziotid")?;
-    let aziottpm_user = crate::internal::common::get_system_user("aziottpm")?;
+    // When run in a snap expect the four users to be prefixed with `snap_`.
+
+    let (aziotks_user, aziotcs_user, aziotid_user, aziottpm_user) = if std::env::var("SNAP").is_ok()
```

Comment:
This is a runtime env var and not a compile-time one?

Reply:
Hmm... I was thinking that doing something compile-time might require a larger change although I can see that from an overall cleanliness perspective it provides benefits.  Let me take a closer look as perhaps it wouldn't be as involved as I assume.

Reply:
You'd replace `if std::env::var("SNAP").is_some()` with `if option_env!("SNAP").is_some()`

The question is more about: Will the snap package be reusing the binaries of the regular Ubuntu build? Is that env var automatically defined for snaps at runtime?

Reply:
The binaries for a regular Ubuntu build should work as well.  Yes, the env var is automatically injected for a snap at runtime by the snapd process.

However, I'm making a change to this PR to have the userid be set at build time.  A motivating reason is to unblock further development by making it simple to change the userid's (e.g. root vs snap_* vs the aziot??) without requiring a code change every time.